GitUI/CommandsDialogs/FormDeleteBranch.cs

```diff
@@ -48,15 +48,22 @@ protected override void OnRuntimeLoad(EventArgs e)
             base.OnRuntimeLoad(e);
 
             Branches.BranchesToSelect = Module.GetRefs(RefsFilter.Heads).ToList();
-            foreach (string branch in Module.GetMergedBranches())
+            if (AppSettings.DontConfirmDeleteUnmergedBranch)
             {
-                if (!branch.StartsWith("* "))
-                {
-                    _mergedBranches.Add(branch.Trim());
-                }
-                else if (!branch.StartsWith("* ") || !DetachedHeadParser.IsDetachedHead(branch[2..]))
+                _currentBranch = Module.GetSelectedBranch();
+            }
+            else
+            {
+                foreach (string branch in Module.GetMergedBranches())
                 {
-                    _currentBranch = branch.Trim('*', ' ');
+                    if (!branch.StartsWith("* "))
+                    {
+                        _mergedBranches.Add(branch.Trim());
+                    }
+                    else if (!branch.StartsWith("* ") || !DetachedHeadParser.IsDetachedHead(branch[2..]))
```

Comment:
Not changed in this PR but still confusing
Also affects comment on line 92
```suggestion
                    else if (!DetachedHeadParser.IsDetachedHead(branch[2..]))
```

Reply:
> Not changed in this PR but still confusing

Indeed, this did not make any sense.

> Also affects comment on line 92

👍 moving this, makes the code clearer